astroquery/nasa_exoplanet_archive/core.py

```diff
@@ -143,14 +181,26 @@ def query_criteria_async(self, table, get_query_payload=False, cache=None, **cri
         if cache is None:
             cache = self.CACHE
 
-        # Execute the request
-        response = self._request(
-            "GET", self.URL, params=request_payload, timeout=self.TIMEOUT, cache=cache,
-        )
-        response.requested_format = criteria["format"]
+        # The _request method is a custom astroquery wrapper around the requests.request function that provides important astroquery-specific utility, including caching, HTTP header generation, progressbars, and local writing-to-disk.
+        # This needs to be updated to use TAP (TapPlus), and the request_payload formatted correctly
+        # Execute the request (generic HTTP request, similar to requests.Session.request but with added caching-related tools)
+        if table in ["ps", "pscomppars"]:
```

Comment:
An issue with this interface (the TAP table names being hard coded) is that as more tables are migrated new versions of astroquery will need to be released. A better way might be to hard code the API table names and have the *default* be TAP. What are the chances that TAP tables with the same names as API tables will be released?

Reply:
Similarly, I think that it might be worth adding a `force_tap` parameter or something to make it possible to override the default behavior unless we're confident that we can cover all the future tables!

Reply:
That sounds sensible. I think the chances that we'll duplicate table names are pretty low (to avoid confusion within the archive too). I'll make this change.

Reply:
Actually it seems some of the table names will remain the same. Could I try given table name against the TAP service, and if it returns an error, try against the API service?

Reply:
I'll make use of: https://exoplanetarchive.ipac.caltech.edu/TAP/sync?query=select+schema_name,table_name+from+TAP_SCHEMA.tables&format=csv

Reply:
Awesome - yeah that would be a great way to do it, I think!

Reply:
Did this:
```
# Tables accessed by API are gradually migrating to TAP service. Generate current list of tables in TAP.
tap = pyvo.dal.tap.TAPService(baseurl=conf.url_tap)
response = tap.search(query="select * from TAP_SCHEMA.tables", language="ADQL")
TAP_TABLES = [table for table in response["table_name"].data if "TAP_SCHEMA." not in table]
```